Sources/Build/BuildPlan.swift

```diff
@@ -712,6 +719,7 @@ public final class SwiftTargetBuildDescription {
         prebuildCommandResults: [PrebuildCommandResult] = [],
         isTestTarget: Bool? = nil,
         isTestDiscoveryTarget: Bool = false,
+        isSynthesizedTestManifestTarget: Bool = false,
```

Comment:
feels like we have too many mutually exclusive booleans here. should this be an enum instead?

Reply:
I have a different idea about how to address this which would obviate the need for this new property — hope to have that done soon once I'm finished with some meetings.

Reply:
Update… My idea did help simplify one thing: I was able to remove a special case in the computed `var needsToBeParsedAsLibrary: Bool` property getter by explicitly configuring the synthesized test manifest target as a `.library` type target since, when we synthesize the code for that target, it always uses an `@main` type so we do want the `-parse-as-library` compiler flag.

But I wasn't able to fully eliminate this trio of test target-related properties, after I remembered there is one other usage (in `addTestManifestGenerationCommand` added in this PR) which still needs to distinguish a _synthesized_ test manifest target from other kinds of test targets, so it can know which target to synthesize the runner code into (if any).

So @tomerd, I ended up taking your suggestion and consolidated these three properties into a new enum type called `TestTargetRole`, and agree it's a bit nicer now overall. Thx for the suggestion!